src/reportSources/legacyFflogs/eventAdapter/__tests__/deduplicateAoE.ts

```diff
@@ -0,0 +1,258 @@
+					...eventAttrs,
+					timestamp: 2,
+					targets: [{
+						target: 't2',
+						...damageTargetAttrs,
+					}],
+				},
+			]
+
+			it('deduplicates damage events', () => {
+				const results = step.postprocess(events)
+
+				// eslint-disable-next-line @typescript-eslint/no-magic-numbers
+				expect(results.length).toEqual(3)
+				expect((results[0] as Events['damage']).targets.length).toEqual(2)
+			})
+			it('does not deduplicate DoT events', () => {
+				const results = step.postprocess(events)
+
+				expect(results.filter(e => (e as Events['damage']).sequence != null).length).toEqual(1)
+			})
+			it('retains order of events', () => {
+				const results = step.postprocess(events)
+
+				ensureOrder(results.map(e => e.timestamp))
+			})
+		})
+
+		describe('when dealing with heal events', () => {
+			const events: Event[] = [
+				{
+					type: 'heal',
+					...eventAttrs,
+					timestamp: 1,
+					sequence: 1,
+					targets: [{
+						target: 't1',
+						...healTargetAttrs,
+					}],
+				},
+				{
+					type: 'heal',
+					...eventAttrs,
+					timestamp: 1,
+					sequence: 1,
+					targets: [{
+						target: 't2',
+						...healTargetAttrs,
+					}],
+				},
+				{
+					type: 'heal',
+					...eventAttrs,
+					timestamp: 2,
+					targets: [{
+						target: 't1',
+						...healTargetAttrs,
+					}],
+				},
+				{
+					type: 'heal',
+					...eventAttrs,
+					timestamp: 2,
+					targets: [{
+						target: 't1',
+						...healTargetAttrs,
+					}],
+				},
+			]
+			it('deduplicates heal events', () => {
+				const results = step.postprocess(events)
+
+				// eslint-disable-next-line @typescript-eslint/no-magic-numbers
+				expect(results.length).toEqual(3)
+				expect((results[0] as Events['heal']).targets.length).toEqual(2)
+			})
+			it('does not deduplicate HoT events', () => {
+				const results = step.postprocess(events)
+
+				expect(results.filter(e => (e as Events['heal']).sequence != null).length).toEqual(1)
+			})
+			it('retains order of events', () => {
+				const results = step.postprocess(events)
+
+				ensureOrder(results.map(e => e.timestamp))
+			})
+		})
+
+		describe('when dealing with neither damage nor heal events', () => {
+			const events: Event[] = [
+				{
+					type: 'execute',
+					timestamp: 1,
+					source: 's1',
+					target: 't1',
+					action: 1,
+					sequence: 1,
+				},
+				{
+					type: 'execute',
+					timestamp: 1,
+					source: 's1',
+					target: 't2',
+					action: 1,
+					sequence: 1,
+				},
+				{
+					type: 'execute',
+					timestamp: 2,
+					source: 's1',
+					target: 't1',
+					action: 1,
+					sequence: 2,
+				},
+			]
+
+			it('does not deduplicate events', () => {
+				const results = step.postprocess(events)
+				// eslint-disable-next-line @typescript-eslint/no-magic-numbers
+				expect(results.length).toEqual(3)
+			})
+			it('retains order of events', () => {
+				const results = step.postprocess(events)
+				ensureOrder(results.map(e => e.timestamp))
```

Comment:
See comments on damage event order test - similar issue here with the two execute events at timestamp 1.